app/views/base.py

```diff
@@ -37,6 +37,93 @@ class BaseView(django.views.View):
         'lang': utils.strip,
     }
 
+    class Env(object):
+        """Class to store environment information used by views and templates.
+        """
+        # pylint: disable=attribute-defined-outside-init
+        # pylint: disable=too-many-instance-attributes
+
+        @property
+        def action(self):
+            """Gets the action ID, an identifier for the page being served."""
+            return self._action
+
+        @action.setter
+        def action(self, value):
+            self._action = value
+
+        @property
+        def charset(self):
+            """Gets the character set being used to serve the page."""
```

Comment:
Optional nit: Character **encoding** would be a more accurate name here (though it is named "charset").